op-node/rollup/driver/sequencer.go

```diff
@@ -47,20 +50,27 @@ type Sequencer struct {
 	nextAction time.Time
 }
 
-func NewSequencer(log log.Logger, cfg *rollup.Config, engine derive.ResettableEngineControl, attributesBuilder derive.AttributesBuilder, l1OriginSelector L1OriginSelectorIface, metrics SequencerMetrics) *Sequencer {
+func NewSequencer(log log.Logger, cfg *rollup.Config, engine derive.ResettableEngineControl, coordinatorClient *client.CoordinatorClient, attributesBuilder derive.AttributesBuilder, l1OriginSelector L1OriginSelectorIface, metrics SequencerMetrics) *Sequencer {
 	return &Sequencer{
-		log:              log,
-		config:           cfg,
-		engine:           engine,
-		timeNow:          time.Now,
-		attrBuilder:      attributesBuilder,
-		l1OriginSelector: l1OriginSelector,
-		metrics:          metrics,
+		log:               log,
+		config:            cfg,
+		engine:            engine,
+		coordinatorClient: coordinatorClient,
+		timeNow:           time.Now,
+		attrBuilder:       attributesBuilder,
+		l1OriginSelector:  l1OriginSelector,
+		metrics:           metrics,
 	}
 }
 
 // StartBuildingBlock initiates a block building job on top of the given L2 head, safe and finalized blocks, and using the provided l1Origin.
 func (d *Sequencer) StartBuildingBlock(ctx context.Context) error {
+	// External coordinator mode (configured by --coordinator.enabled=true): Sequencer requests permission to build
+	// new blocks from the external coordinator by calling RequestBuildingBlock().
+	if d.coordinatorClient != nil && !d.coordinatorClient.RequestBuildingBlock() {
+		return errors.New("failed to request permission for building block from coordinator")
+	}
+
```

Comment:
suggest to add this part in `func (d *Sequencer) RunNextSequencerAction`
and wrap the logic to a single function would be nice

Reply:
Updated at https://github.com/bnb-chain/opbnb/pull/13/commits/17113eba408dd7f7208f9eac785bfc2871de6a1e. PTAL.